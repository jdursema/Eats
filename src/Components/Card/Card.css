.card {
  background-color: white;
  width: 250px;
  padding: 5px; }
  .card .restaurant-name {
    background-color: black;
    color: white;
    padding: 5%; }
