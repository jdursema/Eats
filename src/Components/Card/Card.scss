.card {
  background-color: white;
  min-width: 250px;
  min-height: 350px;
	margin: 15px;
  padding: 5px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  border: 2px solid black; 
  .restaurant-name{
    color: #0F2130;
    padding: 5%;
    font-family: 'Spectral SC', serif;
    font-size: 25px;
    line-height: 1;
    margin: 0;
    padding: 0;
    border-bottom: lightgray 1px solid;
  }
  p{
    color: #0F2130;
    font-family: 'Raleway', sans-serif;
    font-size: 14px;
  }
}

.favorite{
  border: gold 2px solid;
}