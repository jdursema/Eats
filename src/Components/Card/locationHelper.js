import { googleApiKey } from './.googleapikey.js';


export const geolocationFetch = async () => {
  const fetchLocation = await fetch(`https://www.googleapis.com/geolocation/v1/geolocate?key=${googleApiKey}`, {
    method: 'POST',
    headers: {'Content-Type': 'application/json' 
    }
  });
  const fetchResponse = await fetchLocation.json();


  const locationObj = await fetchCityName(fetchResponse.location.lat, fetchResponse.location.lng);

  return await locationObj;
};

const fetchCityName = async (lat, lng) => {
  const fetchCity = await fetch (`https://maps.googleapis.com/maps/api/geocode/json?latlng=${lat},${lng}&key=${googleApiKey}
  `);
  const fetchResponse = await fetchCity.json();

  const responseCity = await fetchResponse.results[4].formatted_address;

  const locationObj = await {name: responseCity, lat: lat, lng: lng};


  return locationObj;
};

