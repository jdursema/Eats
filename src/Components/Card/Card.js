import React from 'react';
import './Card.css';

const Card = ({ info, handleAddFav, user, history, favorited }) => {
  let keys
  let mappedData

  if (info.data){
    keys = Object.keys(info.data);
    console.log(keys)
    mappedData = keys.map((key, index)=> {
      return <p key={index}>{key}: {info.data[key]}</p>;
    });
  }

  const handleFavClick = (info, user) => {
    if (user.uid){
      handleAddFav(info, user)
    } else {
      history.push('/login')
    }
  }
  

  return (
    <div className = 'card'>
      <p className='restaurant-name'>{info.name}</p>
      {mappedData}
      <button onClick={()=> { handleFavClick(info, user)}}>
        Fav
      </button>
    </div>
  ); 
};

export default Card;