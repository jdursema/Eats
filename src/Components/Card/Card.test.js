import Card from './Card';
import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

describe('card', () => {
  it('should match its snapshot', () => {

    const mockFunc = jest.fn()

    const cardObj = {name: 'Scales', data:{
      hours: '9-5',
      reservations: 'no'}
    }

    const card = shallow(<Card
      info= {cardObj}
      handleAddFav = {mockFunc}
      handleDeleteFav = {mockFunc}
      user = {{name: 'Julie'}}
      history = {'hi'}
      favorited = {false}
      favorites={[{name: 'Duck Fat'}]}
     />)

     expect(card).toMatchSnapshot()
  })
})