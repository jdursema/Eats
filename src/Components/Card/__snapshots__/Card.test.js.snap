// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card should match its snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    favorited={false}
    favorites={
        Array [
            Object {
              "name": "Duck Fat",
            },
          ]
    }
    handleAddFav={[Function]}
    handleDeleteFav={[Function]}
    history="hi"
    info={
        Object {
            "data": Object {
              "hours": "9-5",
              "reservations": "no",
            },
            "name": "Scales",
          }
    }
    user={
        Object {
            "name": "Julie",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="restaurant-name"
>
          Scales
</p>,
        Array [
          <p>
            hours
            : 
            9-5
</p>,
          <p>
            reservations
            : 
            no
</p>,
        ],
        <button
          onClick={[Function]}
>
          Fav
</button>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Scales",
          "className": "restaurant-name",
        },
        "ref": null,
        "rendered": "Scales",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "hours",
            ": ",
            "9-5",
          ],
        },
        "ref": null,
        "rendered": Array [
          "hours",
          ": ",
          "9-5",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "reservations",
            ": ",
            "no",
          ],
        },
        "ref": null,
        "rendered": Array [
          "reservations",
          ": ",
          "no",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Fav",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Fav",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="restaurant-name"
>
            Scales
</p>,
          Array [
            <p>
              hours
              : 
              9-5
</p>,
            <p>
              reservations
              : 
              no
</p>,
          ],
          <button
            onClick={[Function]}
>
            Fav
</button>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Scales",
            "className": "restaurant-name",
          },
          "ref": null,
          "rendered": "Scales",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "hours",
              ": ",
              "9-5",
            ],
          },
          "ref": null,
          "rendered": Array [
            "hours",
            ": ",
            "9-5",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "reservations",
              ": ",
              "no",
            ],
          },
          "ref": null,
          "rendered": Array [
            "reservations",
            ": ",
            "no",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Fav",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Fav",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
